index.js

```diff
@@ -262,13 +262,39 @@ module.exports = {
               __frameId: params.frameId,
               _initialPriority: request.initialPriority,
               _priority: request.initialPriority,
-              _initiator: params.initiator.url,
-              _initiator_line: params.initiator.lineNumber,
               pageref: currentPageId,
               request: req,
-              time: 0
+              time: 0,
+              _initiator_detail: JSON.stringify(params.initiator),
+              _initiator_type: params.initiator.type
             };
 
+            // The object initiator change according to its type
+            switch (params.initiator.type) {
+              case 'parser':
+                {
+                  entry._initiator = params.initiator.url;
+                  entry._initiator_line = params.initiator.lineNumber + 1; // Because lineNumber is 0 based
+                }
+                break;
+
+              case 'script':
+                {
+                  if (
+                    params.initiator.stack &&
+                    params.initiator.stack.callFrames.length > 0
+                  ) {
+                    const topCallFrame = params.initiator.stack.callFrames[0];
+                    entry._initiator = topCallFrame.url;
+                    entry._initiator_line = topCallFrame.lineNumber + 1; // Because lineNumber is 0 based
+                    entry._initiator_column = topCallFrame.columnNumber;
```

Comment:
This is also 0-based, right?

Reply:
Idk, will check 😄

Reply:
Yes it's 0 based https://chromedevtools.github.io/devtools-protocol/1-2/Runtime#type-CallFrame

Will commit the fix